storage/src/main/java/com/tencent/rss/storage/handler/impl/LocalFileClientReadHandler.java

```diff
@@ -51,18 +62,44 @@ public LocalFileClientReadHandler(
     this.partitionNum = partitionNum;
     this.readBufferSize = readBufferSize;
     this.shuffleServerClients = shuffleServerClients;
+    this.expectBlockIds = expectBlockIds;
   }
 
-  @Override
-  public ShuffleDataResult readShuffleData(int segmentIndex) {
+  public ShuffleIndexResult readShuffleIndex() {
+    boolean readSuccessful = false;
+    ShuffleIndexResult shuffleIndexResult = null;
+    long indexDataLength = expectBlockIds.getLongCardinality() * FileBasedShuffleSegment.SEGMENT_SIZE;
```

Comment:
index file include expect Blocks and unexpect Blocks. We cannot read all the ndex file.

Reply:
Yes I will restore the position.